apps/portal/src/actions.js

```diff
@@ -482,7 +482,12 @@ async function updateProfile({data, state, api}) {
     } else if (emailUpdate) {
         const action = emailUpdate.success ? 'updateProfile:success' : 'updateProfile:failed';
         const status = emailUpdate.success ? 'success' : 'error';
-        const message = !emailUpdate.success ? t('Failed to send verification email') : t('Check your inbox to verify email update');
+        let message = '';
+        if (emailUpdate.error.message === 'Memberships from this email domain are currently restricted.') {
+            message = chooseBestErrorMessage(emailUpdate.error, t('Memberships from this email domain are currently restricted.'), t);
+        } else {
+            message = emailUpdate.success ? t('Check your inbox to verify email update') : t('Failed to send verification email');
+        }
```

Comment:
I'd suggest not matching for `emailUpdate.error.message === 'Memberships from this email domain are currently restricted.'` here, as this limits our ability to have more error messages in the future.

Instead I'd recommend doing something like:

```
if (emailUpdate.error) {
    message = chooseBestErrorMessage(emailUpdate.error, t('Failed to send verification email'), t);
} else {
   message = t('Check your inbox to verify email update');
}
```

and then add the string `"Memberships from this email domain are currently restricted."` under `setupSpecialMessages` [here](https://github.com/TryGhost/Ghost/blob/a5f8860bcc141a7fa8415013ff2768e6f058e0eb/apps/portal/src/utils/errors.js#L43C1-L65C7)

Reply:
Fixed in 8bd4ee18d9caa85bf5f6ed5ddb7a577311833b86